dlib/gui_widgets/widgets.cpp

```diff
@@ -18,6 +18,9 @@ namespace dlib
 // ----------------------------------------------------------------------------------------
 // ----------------------------------------------------------------------------------------
 
+    // trz
+    int auto_part_num = 0;
```

Comment:
Never ever use a global variable.  This should be a class member.

Reply:
Fixed